agent/connect/ca/provider_vault.go

```diff
@@ -496,13 +497,21 @@ func (v *VaultProvider) SetIntermediate(intermediatePEM, rootPEM string) error {
 		return err
 	}
 
-	_, err = v.writeNamespaced(v.config.IntermediatePKINamespace, v.config.IntermediatePKIPath+"intermediate/set-signed", map[string]interface{}{
+	importResp, err := v.writeNamespaced(v.config.IntermediatePKINamespace, v.config.IntermediatePKIPath+"intermediate/set-signed", map[string]interface{}{
 		"certificate": intermediatePEM,
 	})
 	if err != nil {
 		return err
 	}
 
+	// Vault 1.11+ will return a non-nil response from intermediate/set-signed
```

Comment:
This is lifted verbatim from the former fix.

Reply:
Should we wrap this shared bit of code in a common for DRYness?

Reply:
That already is in `setDefaultIntermediateIssuer`

Reply:
Yeah when I was refactoring `setDefaultIntermediateIssuer` I thought about bringing the nil check inside the function. I concluded that it's more useful to be explicitly loud about some behavioural assumptions than to call `setDefaultIntermediateIssuer` for all versions but secretly have it be a no-op for some versions.